lib/net.c

```diff
@@ -1432,12 +1507,22 @@ net_launch_at_stage1(struct net_config *net,
 	RTE_VERIFY(front_tx_queues >= 0);
 	net->front.num_rx_queues += front_rx_queues;
 	net->front.num_tx_queues += front_tx_queues;
+	net->gatekeeper_num_rx_queues = RTE_MAX(net->gatekeeper_num_rx_queues,
+		net->front.num_rx_queues);
+	net->gatekeeper_num_tx_queues = RTE_MAX(net->gatekeeper_num_tx_queues,
+		net->front.num_tx_queues);
```

Comment:
Reading the code further, I noticed that the code in `init_net_stage1()` cannot go to `start_network_stage2()` because functional blocks need the memory pools to initialize their ports. The solution is to launch `init_net_stage1()` at stage 1 after launching `init_iface_stage1()` for the back interface (see `gatekeeper_init_network()`).